tokio/src/runtime/task/mod.rs

```diff
@@ -184,6 +184,27 @@ use std::marker::PhantomData;
 use std::ptr::NonNull;
 use std::{fmt, mem};
 
+/// An opaque ID that uniquely identifies a task relative to all other currently
+/// running tasks.
+///
+/// # Notes
+///
+/// - Task IDs are unique relative to other *currently running* tasks. When a
+///   task completes, the same ID may be used for another task.
+/// - Task IDs are *not* sequential, and do not indicate the order in which
+///   tasks are spawned, what runtime a task is spawned on, or any other data.
+///
+/// **Note**: This is an [unstable API][unstable]. The public API of this type
+/// may break in 1.x releases. See [the documentation on unstable
+/// features][unstable] for details.
+///
+/// [unstable]: crate#unstable-features
+#[cfg_attr(docsrs, doc(cfg(all(feature = "rt", tokio_unstable))))]
+#[cfg_attr(not(tokio_unstable), allow(unreachable_pub))]
+// TODO(eliza): there's almost certainly no reason not to make this `Copy` as well...
```

Comment:
we should probably just derive `Copy` as well, but i didn't add it out of an abundance of caution for forwards-compatibility. anyone have a problem with deriving `Copy` for these?

Reply:
Doesn't `std` have `Id` actually hold an `Arc`? While i don't think we want to do this, we may want to wait hold off on `Copy`.

Reply:
But, I also don't care that much, if y'all think `Copy` is good, do it.

Reply:
> Doesn't `std` have `Id` actually hold an `Arc`? While i don't think we want to do this, we may want to wait hold off on `Copy`.

Yeah, this is why I didn't add `Copy` to this, so that we could reserve the right to change the repr however we want to later. I just think we are probably not going to actually do that. I'll hold off on making that commitment for now.